imports/plugins/core/orders/client/components/orderDashboard.js

```diff
@@ -26,21 +30,74 @@ class OrderDashboard extends Component {
     toggleShippingFlowList: PropTypes.func
   }
 
-  state = {
-    detailClassName: "",
-    listClassName: "order-icon-toggle",
-    openList: true,
-    orders: this.props.orders,
-    query: this.props.query
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      detailClassName: "",
+      listClassName: "order-icon-toggle",
+      openList: true,
+      orders: this.props.orders,
+      query: this.props.query,
+      searchQuery: ""
+    };
+
+    this.dep = new Tracker.Dependency;
+  }
+
+  componentDidMount() {
+    this.setupTracker();
   }
 
   componentWillReceiveProps(nextProps) {
+    this.setupTracker();
     this.setState({
       orders: nextProps.orders,
       query: nextProps.query
     });
   }
 
+  componentWillUnmount() {
+    this.subscription.stop();
+  }
+
+  // Extracted Tracker logic for the search subscription, to allow calling in both
+  // componentDidMount and componentWillReceiveProps
+  // This tracker is setup in the component because we need to re-subscribe when search input changes
+  setupTracker = () => {
+    Tracker.autorun(() => {
```

Comment:
This should be happening in a container and passed down to the component as props. 
Rule of thumb: Components are presentational

Reply:
Well.... thanks.

I did make a comment just above the function to say why it's being done there. 
Or perhaps you meant that I should just move the same function (as is) into the container file? I'll do that now, but just as we're talking containers/components... the current orderdashboardcontainer probably needs more work to function as HOC really.